utils/fixed_decimal/src/decimal.rs

```diff
@@ -1989,24 +2521,130 @@ impl FixedDecimal {
-                if self.nonzero_magnitude_end() < position - 1 {
-                    true
+        self.half_even_to_increment(position, RoundingIncrement::MultiplesOf1);
+    }
+
+    /// Take the half even of the number at a particular position and rounding increment.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use fixed_decimal::{FixedDecimal, RoundingIncrement};
+    /// # use std::str::FromStr;
+    ///
+    /// let mut dec = FixedDecimal::from_str("-3.5").unwrap();
+    /// dec.half_even_to_increment(0, RoundingIncrement::MultiplesOf2);
+    /// assert_eq!("-4", dec.to_string());
+    /// let mut dec = FixedDecimal::from_str("7.57").unwrap();
+    /// dec.half_even_to_increment(-1, RoundingIncrement::MultiplesOf5);
+    /// assert_eq!("7.5", dec.to_string());
+    /// let mut dec = FixedDecimal::from_str("5.45").unwrap();
+    /// dec.half_even_to_increment(-2, RoundingIncrement::MultiplesOf25);
+    /// assert_eq!("5.50", dec.to_string());
+    /// let mut dec = FixedDecimal::from_str("9.99").unwrap();
+    /// dec.half_even_to_increment(-1, RoundingIncrement::MultiplesOf25);
+    /// assert_eq!("10.0", dec.to_string());
+    /// let mut dec = FixedDecimal::from_str("9.99").unwrap();
+    /// dec.half_even_to_increment(-2, RoundingIncrement::MultiplesOf2);
+    /// assert_eq!("9.98", dec.to_string());
+    /// ```
+    pub fn half_even_to_increment(&mut self, position: i16, increment: RoundingIncrement) {
+        let should_expand = match increment {
+            RoundingIncrement::MultiplesOf1 => {
+                let digit_after_position = self.digit_at_next_position(position);
+                match digit_after_position.cmp(&5) {
+                    Ordering::Less => false,
+                    Ordering::Greater => true,
+                    Ordering::Equal => {
+                        // NOTE: `digit_after_position` equals to 5, this means that `position`
+                        // does not equal i16::MIN.
+                        if self.nonzero_magnitude_end() < position - 1 {
+                            true
+                        } else {
+                            self.digit_at(position) % 2 != 0
+                        }
+                    }
+                }
+            }
+            RoundingIncrement::MultiplesOf2 => {
+                let current_digit = self.digit_at(position);
+
+                // Equivalent to "if current_digit is odd".
+                if (current_digit & 0x01) == 1 {
+                    if self.nonzero_magnitude_end() < position {
+                        true
+                    } else {
+                        // This essentially expands to the "even" increments,
+                        // or the increments that are in the even places on the
+                        // rounding range: [0, 4, 8]
+                        // Equivalent to `(current_digit / 2) is odd`.
+                        (current_digit >> 1) & 0x01 == 1
```

Comment:
Technically you need to look at two digits because this is periodic every 20 not every 10:

- 1 => down to 0
- 3 => up to 4
- 5 => down to 4
- 7 => up to 8
- 9 => down to 8
- 11 => up to 12
- 13 => down to 12
- 15 => up to 16
- 17 => down to 16
- 19 => up to 20
- 21 => down to 20
- ...